datastore/datastore.go

```diff
@@ -115,6 +118,7 @@ func NewClientWithDatabase(ctx context.Context, projectID, databaseID string, op
 	} else {
 		o = []option.ClientOption{
 			option.WithEndpoint(prodAddr),
```

Comment:
Can we use withDefaultEndpoint() in place of WithEndpoint()?
The WithEndpoint() param is treated as an overriding option by the transport layer code, thus preventing the WithDefaultMTLSEndpoint() param from being used. thx

Reply:
Compiler warning:

internaloption.WithDefaultEndpoint is deprecated: WithDefaultEndpoint does not support setting the universe domain. Use WithDefaultEndpointTemplate and WithDefaultUniverseDomain to compose the default endpoint instead.

So, used template and domain

Reply:
I see looks like things changed recently, but yes you are right, we can use the template and domain instead, e.g. https://github.com/googleapis/google-cloud-go/blob/main/ai/generativelanguage/apiv1/generative_client.go#L61